src/main/java/edu/nitt/delta/bustracker/controller/api/VehicleController.java

```diff
@@ -117,4 +117,24 @@
                     HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
+    @PostMapping("/{id}/status")
+    public ResponseEntity<VehicleResponse> changeVehicleStatus(
+            @PathVariable String id, @RequestBody Vehicle vehicle) {
+        try {
+            Vehicle updatedVehicle =
+                    vehicleService.changeVehicleStatus(id, vehicle.getIsOccupied());
+
+            VehicleResponse res =
+                    VehicleResponse.builder().vehicle(updatedVehicle).message("OK").build();
+
+            return new ResponseEntity<>(res, HttpStatus.OK);
```

Comment:
So when you're updating the location object's isOccupied, check if the currently logged in driver's id matches with the driverId of that document, else send a 401